org.envirocar.app/src/org/envirocar/app/views/tracklist/TrackListLocalCardFragment.java

```diff
@@ -387,6 +391,13 @@ public void onNext(Track track) {
         @Override
         public void onError(Throwable e) {
             LOG.warn(e.getMessage(), e);
+            if (!isNetworkAvailable() && getView()!=null) {
+                Snackbar.make(getView(),
+                        R.string.track_list_upload_error_no_network_connection,
+                        Snackbar.LENGTH_LONG)
+                        .show();
+            }
```

Comment:
I'd suggest, it would be better to check internet connection before the app tries to upload the tracks. This will prevent the app to run into a error response that have to be handled. Server response errors still can be handled separately within `onError()`.

Reply:
>I'd suggest, it would be better to check the internet connection before the app tries to upload the tracks

This will be indeed better to check earlier, will update the PR soon :)
maybe we can add checks the same way other than the server errors? maybe ``NOT_ENOUGH_MEASUREMENTS`` error